hnn_core/network.py

```diff
@@ -271,78 +464,38 @@ def _instantiate_feeds(self, n_trials=1):
-                event_times = list()
-                if src_type == 'common':
-                    for idx, gid in enumerate(self.gid_ranges['common']):
-                        gid_target = None  # 'common' attaches to all
-                        event_times.append(
-                            feed_event_times(
-                                feed_type=src_type,
-                                target_cell_type=gid_target,
-                                params=p_common[idx],
-                                gid=gid)
-                        )
-
-                elif src_type in p_unique.keys():
-                    for gid in self.gid_ranges[src_type]:
-                        gid_target = gid - self.gid_ranges[src_type][0]
-                        target_cell_type = self.gid_to_type(gid_target)
-                        event_times.append(
-                            feed_event_times(
-                                feed_type=src_type,
-                                target_cell_type=target_cell_type,
-                                params=p_unique[src_type],
-                                gid=gid)
-                        )
-                feed_times[src_type].append(event_times)
-
-        self.feed_times = feed_times
-
-        # Add tonic inputs from param files
-        self.feed_times['tonic'] = dict()
-        for cell_type in ['L2Basket', 'L5Basket', 'L2Pyr', 'L5Pyr']:
-            is_tonic_present = [f'Itonic_{p}_{cell_type}_soma' in self.params
-                                for p in ['A', 't0', 'T']]
-            if any(is_tonic_present):
-                if not all(is_tonic_present):
-                    raise ValueError(f'Tonic input must have the amplitude, '
-                                     f'start time and end time specified. One '
-                                     f'or more parameter may be missing for '
-                                     f'cell type {cell_type}')
-                self.add_tonic_input(
-                    cell_type=cell_type,
-                    amplitude=self.params[f'Itonic_A_{cell_type}_soma'],
-                    t0=self.params[f'Itonic_t0_{cell_type}_soma'],
-                    T=self.params[f'Itonic_T_{cell_type}_soma']
-                )
-
-    def add_tonic_input(self, cell_type, amplitude, t0, T):
-        """Create tonic feed for a given cell type.
+            for drive_name in self.external_drives.keys():
+                event_times = list()  # new list for each trial and drive
+
+                dyn_specs = self.external_drives[drive_name]['dynamics']
+                drive_type = self.external_drives[drive_name]['type']
+                seedcore = self.external_drives[drive_name]['seedcore']
+
+                # loop over drive 'cells' and create event times for each
+                for drive_cell in self.external_drives[drive_name]['cells']:
+                    event_times.append(
+                        drive_cell_event_times(drive_type, drive_cell,
```

Comment:
honestly I'd rather not call this function and directly use the low level functions. But we can leave it for another PR